prover/src/zk_circuits_handler/bernoulli.rs

```diff
@@ -70,34 +69,50 @@ impl BaseCircuitsHandler {
                 self.get_output_dir(),
             )?;
 
-            return serde_json::to_string(&chunk_proof).map_err(|e| anyhow::anyhow!(e));
+            return Ok(chunk_proof);
         }
         unreachable!("please check errors in proof_type logic")
     }
 
-    fn gen_batch_proof(&self, task: &crate::types::Task) -> Result<String> {
-        let chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)> =
-            self.gen_chunk_hashes_proofs(task)?;
-        let chunk_proofs: Vec<ChunkProof> =
-            chunk_hashes_proofs.iter().map(|t| t.1.clone()).collect();
+    fn gen_chunk_proof(&self, task: &crate::types::Task) -> Result<String> {
+        let chunk_trace = self.gen_chunk_traces(task)?;
+        let chunk_proof = self.gen_chunk_proof_raw(chunk_trace)?;
+        serde_json::to_string(&chunk_proof).map_err(|e| anyhow::anyhow!(e))
+    }
 
+    fn gen_batch_proof_raw(
+        &self,
+        chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)>,
+    ) -> Result<BatchProof> {
         if let Some(prover) = self.batch_prover.as_ref() {
+            let chunk_proofs: Vec<ChunkProof> =
+                chunk_hashes_proofs.iter().map(|t| t.1.clone()).collect();
```

Comment:
why need `clone` here?

Reply:
the chunk_hashes_proofs will be passed to gen_agg_evm_proof function, so the chunk_proofs part need to be cloned here